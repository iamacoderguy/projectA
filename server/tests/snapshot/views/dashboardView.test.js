
const pug = require('pug');
const renderPug = data => pug.renderFile('views/dashboardView.pug', data);

describe('views/dashboardView', () => {
    it('should render dashboardView correctly', () => {
        expect(renderPug({
            path: 'path-to-files',
            dashboardName: 'dashboard-name'
          })).toMatchSnapshot();
    });
});