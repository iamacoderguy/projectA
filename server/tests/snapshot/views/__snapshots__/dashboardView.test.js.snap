// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboardView should render dashboardView correctly 1`] = `"<!DOCTYPE html><html lang=\\"en-US\\"><head><title>dashboard-name</title><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"><meta name=\\"description\\" content=\\"dashboard-name\\"><link rel=\\"stylesheet\\" href=\\"w3.css\\"></head><body><div class=\\"w3-container\\"><div class=\\"w3-container w3-dark-teal\\"><h2>dashboard-name</h2></div><p><iframe width=\\"100%\\" height=\\"200vw\\" title=\\"Result box\\" srcdoc=\\"&lt;i&gt;Result...&lt;/i&gt;\\" name=\\"iframe_result\\"></iframe></p><form class=\\"w3-container\\" method=\\"POST\\" action=\\"/api/files/path?_method=PUT\\" target=\\"iframe_result\\"><p><label class=\\"w3-text-dark-teal\\" for=\\"sharedPathInput\\"><b>Set a new shared path (/api/files/path)</b></label><input class=\\"w3-input w3-border w3-round\\" id=\\"sharedPathInput\\" name=\\"path\\" type=\\"text\\" placeholder=\\"path-to-files\\" value=\\"path-to-files\\"></p><p><button class=\\"w3-btn w3-dark-teal\\" type=\\"submit\\">PUT</button></p></form><form class=\\"w3-container\\" method=\\"GET\\" action=\\"/api/files\\" target=\\"iframe_result\\"><p><label class=\\"w3-text-dark-teal\\"><b>Get the shared files list (/api/files)</b></label></p><p><button class=\\"w3-btn w3-dark-teal\\" type=\\"submit\\">GET</button></p></form><form class=\\"w3-container\\" method=\\"GET\\" action=\\"/api/files\\" target=\\"_blank\\"><p><label class=\\"w3-text-dark-teal\\" for=\\"filenameInput\\"><b>Get a shared file (/api/files/:filename)</b></label><input class=\\"w3-input w3-border w3-round\\" id=\\"filenameInput\\" name=\\"filename\\" type=\\"text\\" label=\\"filenameInput\\" placeholder=\\"filename\\"></p><p><button class=\\"w3-btn w3-dark-teal\\" type=\\"submit\\">GET</button></p></form></div></body></html>"`;
